pootle/apps/pootle_config/managers.py

```diff
@@ -0,0 +1,149 @@
+        if model:
+            self.query_model = model
+        elif hasattr(self, "query_model"):
+            model = self.query_model
+        if model is None:
+            return {}
+        ct = ContentType.objects.get_for_model(model)
+        kwargs = dict(content_type=ct)
+        if isinstance(model, models.Model):
+            kwargs["object_pk"] = force_text(model._get_pk_val())
+        return kwargs
+
+    @property
+    def base_qs(self):
+        qs = self.__class__(self.model)
+        if hasattr(self, "query_model"):
+            qs.query_model = self.query_model
+        return qs
+
+    def config_for(self, model):
+        """
+        QuerySet for all comments for a particular model (either an instance or
+        a class).
+        """
+        self.query_model = model
+        return self.base_qs.filter(**self.get_model_kwargs(model))
+
+    def site_config(self):
+        self.query_model = None
+        return self.base_qs.filter(
+            content_type__isnull=True, object_pk__isnull=True)
+
+    def get_query_model(self, model=None):
+        if model:
+            self.query_model = model
+        elif hasattr(self, "query_model"):
+            model = self.query_model
+        return model
+
+    def should_append_config(self, key, value, model=None):
+        if isinstance(model, models.Model):
+            no_append = config_should_not_be_appended.get(
+                model.__class__,
+                instance=model,
+                key=key, value=value)
+        else:
+            no_append = config_should_not_be_appended.get(
+                model,
+                key=key, value=value)
+        if no_append:
+            raise ConfigurationError(no_append)
+        return True
+
+    def should_set_config(self, key, value, model=None):
+        if isinstance(model, models.Model):
+            no_set = config_should_not_be_set.get(
+                model.__class__,
+                instance=model,
+                key=key, value=value)
+        else:
+            no_set = config_should_not_be_set.get(
+                model,
+                key=key, value=value)
+        if no_set:
+            raise ConfigurationError(no_set)
+        return True
+
+    def append_config(self, key, value="", model=None):
+        model = self.get_query_model(model)
+        if self.should_append_config(key, value, model):
+            return self.create(
+                key=key, value=value,
+                **self.get_model_kwargs(model))
+
+    def clear_config(self, key, model=None):
+        self.get_config_queryset(model).filter(key=key).delete()
+
+    def set_config(self, key, value="", model=None):
+        model = self.get_query_model(model)
+        conf = self.get_config_queryset(model)
+        if self.should_set_config(key, value, model):
+            try:
+                conf = conf.get(key=key)
+            except conf.model.DoesNotExist:
+                conf = self.create(
```

Comment:
@phlax it seemed that you called `self.objects.create()` and I thought there is extra query. So I was wrong I see now that you just create an object here.
But probably if you call  `self.objects.create()` with ready value the code would be less branchy.